src/Plugin/CookiePlugin.php

```diff
@@ -69,7 +69,7 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
                     }
 
                     // Restrict setting cookie from another domain
-                    if (false === strpos($cookie->getDomain(), $request->getUri()->getHost())) {
+                    if (false === strpos($request->getUri()->getHost(), $cookie->getDomain())) {
```

Comment:
i think that is the right approach. afaik the server could send the leading `.` already, we should handle that properly as well, not that we end up with "..symfony.slack.com".

are you sure that cookies should propagate down several levels? like test.com cookie being sent to foo.bar.baz.test.com ?